src/netlink_decoder_ipset.hrl

```diff
@@ -0,0 +1,328 @@
+%% ============================
+
+decode_nl_msgtype_ipset(Id) ->
+    Mapping = #{
+        ?IPSET_CMD_NONE => none,
+        ?IPSET_CMD_PROTOCOL => protocol,
+        ?IPSET_CMD_CREATE => create,
+        ?IPSET_CMD_DESTROY => destroy,
+        ?IPSET_CMD_FLUSH => flush,
+        ?IPSET_CMD_RENAME => rename,
+        ?IPSET_CMD_SWAP => swap,
+        ?IPSET_CMD_LIST => list,
+        ?IPSET_CMD_SAVE => save,
+        ?IPSET_CMD_ADD => add,
+        ?IPSET_CMD_DEL => del,
+        ?IPSET_CMD_TEST => test,
+        ?IPSET_CMD_HEADER => header,
+        ?IPSET_CMD_TYPE => type,
+        ?IPSET_CMD_RESTORE => restore,
+        ?IPSET_CMD_HELP => help,
+        ?IPSET_CMD_VERSION => version,
+        ?IPSET_CMD_QUIT => quit,
+        ?IPSET_CMD_COMMIT => commit
+    },
+    maps:get(Id, Mapping, Id).
+
+%% ============================
+
+decode_ipset(Family, Id, Value, List) ->
+    Type = Id band (bnot (?NLA_F_NET_BYTEORDER bor ?NLA_F_NESTED)),
+    case lists:keyfind(Type, 1, List) of
+        {Type, Key, none} ->
+            {Key, decode_none(Value)};
+        {Type, Key, binary} ->
+            {Key, decode_binary(Value)};
+        {Type, Key, string} ->
+            {Key, decode_string(Value)};
+        {Type, Key, protocol} ->
+            {Key, decode_protocol(Value)};
+        {Type, Key, uint8} ->
+            {Key, decode_uint8(Value)};
+        {Type, Key, uint16}
+                when Id band ?NLA_F_NET_BYTEORDER > 0 ->
+            {Key, decode_uint16(Value)};
+        {Type, Key, uint16} ->
+            {Key, decode_huint16(Value)};
+        {Type, Key, uint32}
+                when Id band ?NLA_F_NET_BYTEORDER > 0 ->
+            {Key, decode_uint32(Value)};
+        {Type, Key, uint32} ->
+            {Key, decode_huint32(Value)};
+        {Type, Key, uint64}
+                when Id band ?NLA_F_NET_BYTEORDER > 0 ->
+            {Key, decode_uint64(Value)};
+        {Type, Key, uint64} ->
+            {Key, decode_huint64(Value)};
+        {Type, Key, Fun}
+                when is_function(Fun, 1) ->
+            {Key, Fun(Value)};
+        {Type, Key, Fun}
+                when is_function(Fun, 3) ->
+            {Key, nl_dec_nla(Family, Fun, Value)};
+        false ->
+            {Type, Value}
+    end.
+
+%% ============================
+
+decode_ipset_attrs(Family, MsgType, Id, Value) ->
+    decode_ipset(Family, Id, Value, [
+        {0, unspec, none},
+        {1, protocol, uint8},
+        {2, setname, string},
+        {3, typename, string},
+        {4, revision, uint8},
+        {5, family, fun (V) -> family_to_atom(decode_uint8(V)) end},
+        {6, flags, uint32}, % TODO: add ipset flags
```

Comment:
I know it got merged already, but I have some questions: 

- shouldn't those TODO have been addressed before merge? 
- On the case, the last question is answered with no. Will a client of this library (ourselves even) run into problems because we assume full implementation of ipset, and there are some things TODO? Should we document those caveats on the README.md?

Reply:
It's more like good to have, than really necessary, e.g. dcos-net PR already uses `flags` as a bit mask with no issue. It doesn't affects anything else. Regarding `skbmark` or `skbprio`, I don't think we will ever use the,, so it doesn't worse the effort to reverse engineer the protocol in that case.